src/main/java/formflow/library/FormFlowController.java

```diff
@@ -21,4 +29,27 @@ protected void saveToRepository(Submission submission, String subflowName) {
     submissionRepositoryService.removeSubflowCSRF(submission, subflowName);
     submissionRepositoryService.save(submission);
   }
+
+  protected FlowConfiguration getFlowConfigurationByName(String flow) {
+      List<FlowConfiguration> flowConfigurationList = flowConfigurations.stream().filter(
+          flowConfiguration -> flowConfiguration.getName().equals(flow)).toList();
+      
+      if (flowConfigurationList.isEmpty()) {
+        throwNotFoundError(flow, null, String.format("Could not find flow %s in your applications flow configuration file.", flow));
+      }
+      
+      return flowConfigurationList.get(0);
+  }
+
+  protected Boolean doesFlowExist(String flow) {
+    return flowConfigurations.stream().anyMatch(
+        flowConfiguration -> flowConfiguration.getName().equals(flow)
+    );
+  }
+
+  protected static void throwNotFoundError(String flow,String screen, String message) {
+    throw new ResponseStatusException(HttpStatus.NOT_FOUND, String.format("There was a problem with the request (flow: %s, screen: %s): %s",
+        flow, screen, message));
+
+  }
```

Comment:
Glad to see these moved out of `ScreenController`! The big refactor I started did something similar for the first two methods so I'll have to reconcile that whenever I can get around to finishing it 😅

Reply:
Would be interested to potentially pair on that when we can prioritize it.